pkg/reconciler/tektontrigger/tektontrigger.go

```diff
@@ -0,0 +1,140 @@
+	tektontriggerreconciler "github.com/tektoncd/operator/pkg/client/injection/reconciler/operator/v1alpha1/tektontrigger"
+	"github.com/tektoncd/operator/pkg/reconciler/common"
+	"knative.dev/pkg/logging"
+	pkgreconciler "knative.dev/pkg/reconciler"
+)
+
+// Reconciler implements controller.Reconciler for Knativeserving resources.
+type Reconciler struct {
+	// kubeClientSet allows us to talk to the k8s for core APIs
+	kubeClientSet kubernetes.Interface
+	// operatorClientSet allows us to configure operator objects
+	operatorClientSet clientset.Interface
+	// manifest is empty, but with a valid client and logger. all
+	// manifests are immutable, and any created during reconcile are
+	// expected to be appended to this one, obviating the passing of
+	// client & logger
+	manifest mf.Manifest
+	// Platform-specific behavior to affect the transform
+	extension common.Extension
+
+	pipelineInformer pipelineinformer.TektonPipelineInformer
+}
+
+// Check that our Reconciler implements controller.Reconciler
+var _ tektontriggerreconciler.Interface = (*Reconciler)(nil)
+var _ tektontriggerreconciler.Finalizer = (*Reconciler)(nil)
+
+// FinalizeKind removes all resources after deletion of a TektonTriggers.
+func (r *Reconciler) FinalizeKind(ctx context.Context, original *v1alpha1.TektonTrigger) pkgreconciler.Event {
+	logger := logging.FromContext(ctx)
+
+	// List all TektonTriggers to determine if cluster-scoped resources should be deleted.
+	tps, err := r.operatorClientSet.OperatorV1alpha1().TektonTriggers().List(metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to list all TektonTriggers: %w", err)
+	}
+
+	for _, tp := range tps.Items {
+		if tp.GetDeletionTimestamp().IsZero() {
+			// Not deleting all TektonPipelines. Nothing to do here.
+			return nil
+		}
+	}
+
+	if err := r.extension.Finalize(ctx, original); err != nil {
+		logger.Error("Failed to finalize platform resources", err)
+	}
+	logger.Info("Deleting cluster-scoped resources")
+	manifest, err := r.installed(ctx, original)
+	if err != nil {
+		logger.Error("Unable to fetch installed manifest; no cluster-scoped resources will be finalized", err)
+		return nil
+	}
+	if err := common.Uninstall(manifest); err != nil {
+		logger.Error("Failed to finalize platform resources", err)
+	}
+	return nil
+}
+
+// ReconcileKind compares the actual state with the desired, and attempts to
+// converge the two.
+func (r *Reconciler) ReconcileKind(ctx context.Context, tt *v1alpha1.TektonTrigger) pkgreconciler.Event {
+	logger := logging.FromContext(ctx)
+	tt.Status.InitializeConditions()
+	tt.Status.ObservedGeneration = tt.Generation
+
+	logger.Infow("Reconciling TektonTriggers", "status", tt.Status)
+
+	//find the valid tekton-pipeline installation
+	if _, err := common.PipelineReady(r.pipelineInformer); err != nil {
+		if err.Error() == common.PipelineNotReady {
+			tt.Status.MarkDependencyInstalling("tekton-pipelines is still installing")
```

Comment:
Should this refer to `tekton-trigger` instead of `tekton-pipeline` ? :+1:

Reply:
i think it should be `tekton-pipelines` because we are making sure pipelines should be running before we start installing triggers.

So for triggers `pipeline` is the dependency and when its not ready we can add a message on triggers status that `pipeline` is still installing

WDYT?

Reply:
Ahhh I see, sounds good :+1: